tests/e2e/configurer/chain/chain.go

```diff
@@ -85,6 +77,20 @@ func (c *Config) CreateNode(initNode *initialization.Node) *NodeConfig {
 	return nodeConfig
 }
 
+// CreateNodeTemp returns new initialized NodeConfig and appends it to a separate list of temporary nodes.
+// This is used for nodes that are intended to only exist for a single test. Without this separation,
+// parallel tests will try and use this node and fail.
+func (c *Config) CreateNodeTemp(initNode *initialization.Node) *NodeConfig {
+	nodeConfig := &NodeConfig{
+		Node:             *initNode,
+		chainId:          c.Id,
+		containerManager: c.containerManager,
+		t:                c.t,
+	}
+	c.NodeTempConfigs = append(c.NodeTempConfigs, nodeConfig)
+	return nodeConfig
+}
```

Comment:
Can 2 parallel tests append and remove concurrently? If so, how do we handle that? Not seeing any mutexes here

Reply:
Only one test (the state sync test) spins up an additional node that we don't want to use across other tests. Agreed this would become problematic if a new test needed to spin up a temp node